File: apps/external-apis/continente-api/app/Application/UseCases/SearchProducts/SearchProductsInput.php
<?php


namespace App\Application\UseCases\SearchProducts;

use App\Core\DTO;

class SearchProductsInput extends DTO
{
    public ?string $query = null;
    public int $page = 50;

    public function __construct(array $data) {
        parent::__construct($data);
    }

    public function query(): ?string
    {
        if ($this->query === null) {
            return null;
        }

        $qry = trim($this->query);
        return $qry === '' ? null : $qry;
    }

    public function page(): int
    {
        return $this->page;
    }
}

File: apps/external-apis/lidl-api/src/Lidl.Application/UseCases/SearchProducts/SearchProductsInput.cs
namespace Lidl.Application.UseCases.SearchProducts;

public sealed class SearchProductsInput
{
    public string? Query { get; set; }
    public int Page { get; set; } = 1;
    public int Limit { get; set; } = 20;

    public SearchProductsInput(string? query, int page, int limit)
    {
        Query = query;
        Page = page;
        Limit = limit;
    }
}